Source/Scene/ModelAnimationCollection.js

```diff
@@ -426,7 +440,17 @@ ModelAnimationCollection.prototype.update = function (frameState) {
     const play =
       (pastStartTime || (repeat && !defined(scheduledAnimation.startTime))) &&
       (delta <= 1.0 || repeat) &&
-      (!defined(stopTime) || JulianDate.lessThanOrEquals(sceneTime, stopTime));
+      !pastStopTime;
+
+    if (
+      delta === scheduledAnimation._prevAnimationTime &&
+      !play === (scheduledAnimation._state === ModelAnimationState.STOPPED)
```

Comment:
I'm changing it to a nested if, to avoid computing `animationStopped` when delta changed (which is presumably the common case, and certainly the case where we care about performance).